src/Model/Documentable/DocumentableProductTrait.php

```diff
@@ -106,23 +91,27 @@ protected function addChannelsInDocument(ResultInterface $document): ResultInter
         return $document;
     }
 
-    /**
-     * @param ResultInterface $document
-     *
-     * @return ResultInterface
-     */
     protected function addPricesInDocument(ResultInterface $document): ResultInterface
     {
         /** @var Channel $channel */
         foreach ($this->getChannels() as $channel) {
-            /** @var ProductVariant $variant */
-            if ($variant = $this->getCheapestVariantForChannel($channel)) {
-                $price = $variant->getChannelPricingForChannel($channel);
-
+            $cheapestVariantChannelPricing = $this->getCheapestVariantChannelPricingForChannel($channel);
+            if ($cheapestVariantChannelPricing !== null) {
                 /** @var CurrencyInterface $currency */
                 foreach ($channel->getCurrencies() as $currency) {
-                    $document->addPrice($channel->getCode(), $currency->getCode(), $price->getPrice());
-                    if ($originalPrice = $price->getOriginalPrice()) {
+                    $variantPrice = min($cheapestVariantChannelPricing->getPrice(), $cheapestVariantChannelPricing->getMinimumPrice());
```

Comment:
Why using the minimum between the price and the minimum price?
As stated [here](https://github.com/Sylius/Sylius/blob/1.12/docs/book/products/pricing.rst#minimum-price) the minimum price is only a "limit" below which the price of the product cannot go while applying promotions . It's not something that should be used outside the logic that applies promotions.

Here we are populating the document with data that must be indexed and i think that the "->getPrice()" is the only thing that should be indexed. It's already limited to the "minimum price" if the promotions that have been applied made the price go below the "minimum price".

Or am i missing something?

Example:
Product has 1€ as minimum price and 10€ as price and 15€ as original price. With the current logic we will index 1€ as the price of the product when it should be 10€.

Reply:
You are right @LucaGallinari. It was right the previous version before the suggested changes: https://github.com/webgriffe/SyliusSearchPlugin/pull/1#discussion_r934251976.

Reply:
Now should be ok!